src/internal/user_api/mod.rs

```diff
@@ -320,6 +343,53 @@ pub async fn user_rotate_private_key<CR: rand::CryptoRng + rand::RngCore>(
     .into())
 }
 
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct DeviceAddResult {
+    /// The user's given id, which uniquely identifies them inside the segment.
+    pub(crate) account_id: UserId,
```

Comment:
These were made `pub(crate)` so that the values can be accessed from the `impl From<DeviceAddResult> for DeviceContext`, which is in `ironoxide::internal`

Reply:
I think it would be somewhat cleaner just to move the `From` into the user_api module. We still need to think about potentially opening up all our *Result types, making all members pub.